src/console_gui.cpp

```diff
@@ -68,9 +69,52 @@ static std::deque<IConsoleLine> _iconsole_buffer;
 
 static bool TruncateBuffer();
 
+class ConsoleAutoCompletion final : public AutoCompletion {
+public:
+	using AutoCompletion::AutoCompletion;
+
+private:
+	std::vector<std::string> GetSuggestions(std::string_view prefix, std::string_view query) override
+	{
+		while (prefix.ends_with(" ")) {
+			prefix.remove_suffix(1);
+		}
```

Comment:
Do we have a "trim" string function already? Seems like it would be more efficient than the explicit loop like that

Reply:
I looked at string_func.h and there are two trim functions:
One for std::string and one for a null-terminated utf-8 strings.
Both of them doesn't combine well with std::string_view (as they can be not null terminated).

I can add one more for std::string_view.
Something like std::string_view StrTrimView(std::string_view s).
It also should be possible to implement StrTrimInPlace as (s = StrTrimView(s)).

Reply:
Implemented my suggestion and added unit tests.
Please let me know if you want me to extract it into a separate commit.